pkg/controller/generate-registry/generate.go

```diff
@@ -285,3 +287,45 @@ func checkSLSAProvenance(assetName, tagName string) *registry.SLSAProvenance {
 		Asset: &assetInfo.Template,
 	}
 }
+
+func checkChecksumCosign(pkgInfo *registry.PackageInfo, checksumAssetName string, assetNames map[string]struct{}) *registry.Cosign {
+	var signatureAssetName string
+	for _, suf := range []string{"-keyless.sig", ".sig"} {
+		if _, ok := assetNames[checksumAssetName+suf]; ok {
+			signatureAssetName = checksumAssetName + suf
+			break
+		}
+	}
+	if signatureAssetName == "" {
+		return nil
+	}
+	var certificateAssetName string
+	for _, suf := range []string{"-keyless.pem", ".pem"} {
+		if _, ok := assetNames[checksumAssetName+suf]; ok {
+			certificateAssetName = checksumAssetName + suf
+			break
+		}
+	}
+	if certificateAssetName == "" {
+		return nil
+	}
+
+	downloadURL := fmt.Sprintf("https://github.com/%s/%s/releases/download/{{.Version}}/",
+		pkgInfo.RepoOwner, pkgInfo.RepoName)
+	return &registry.Cosign{
+		Opts: []string{
+			"--certificate-identity-regexp",
+			fmt.Sprintf(
+				`^https://github\.com/%s/%s/\.github/workflows/.+\.ya?ml@refs/tags/`,
```

Comment:
I left `{{.Version}}` out from here, mostly because I feel simply having the version replaced here literally by templating might not be a good thing to do as the version could contain regex metacharacters.

We could go with this though, on a second look I think I'd actually like this better. Thoughts?
```suggestion
				`^https://github\.com/%s/%s/\.github/workflows/.+\.ya?ml@refs/tags/\Q{{.Version}\E$`,
```